### PR TITLE
feat: add disable AirControl option

### PR DESCRIPTION
**Pull request by BoltSpace**
## Description

Added disable AirControl option to config and function

## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

File: client/main.lua [hunk 1]
```diff
@@ -386,6 +386,18 @@ local function preventVehicleFlip()
     end
 end
 
+local function preventAirControl()
+    local veh = GetVehiclePedIsIn(cache.ped, false)
```
Comment (BerkieBb) [inline client/main.lua]: Use cache.vehicle

File: client/main.lua [hunk 2]
```diff
@@ -386,6 +386,18 @@ local function preventVehicleFlip()
...
+    if DoesEntityExist(veh) and not IsEntityDead(veh) then
```
Comment (BerkieBb) [inline client/main.lua]: You can invert this if statement

File: client/main.lua [hunk 3]
```diff
@@ -386,6 +386,18 @@ local function preventVehicleFlip()
...
+        local model = GetEntityModel(veh)
+        -- If it's not a boat, plane or helicopter, and the vehicle is off the ground with ALL wheels, then block steering/leaning left/right/up/down.
+        if not IsThisModelABoat(model) and not IsThisModelAHeli(model) and not IsThisModelAPlane(model) and IsEntityInAir(veh) then
```
Comment (BerkieBb) [inline client/main.lua]: Same here, invertion

File: config.lua [hunk 4]
```diff
@@ -66,6 +66,7 @@ cfg = {
     limpModeMultiplier = 0.15,					-- The torque multiplier to use when vehicle is limping. Sane values are 0.05 to 0.25
 
     preventVehicleFlip = true,					-- If true, you can't turn over an upside down vehicle
+    preventAirControl = true,					    -- If true, you can't AirControl vehicle
```
Comment (BerkieBb) [inline config.lua]: indent slightly off at the comment

File: client/main.lua (hunk 2 above)
Comment (BoltSpace) [reply client/main.lua]: maybe i'm dumb but just like this?
if not (DoesEntityExist(veh) and not IsEntityDead(veh)) then

Comment (BerkieBb) [reply client/main.lua]: Nope, hold on let me make a suggestion instead

File: client/main.lua [hunk 5]
```diff
@@ -386,6 +386,18 @@ local function preventVehicleFlip()
...
+    local veh = GetVehiclePedIsIn(cache.vehicle, false)
+    if not (DoesEntityExist(veh) and not IsEntityDead(veh)) then
+        local model = GetEntityModel(veh)
+        -- If it's not a boat, plane or helicopter, and the vehicle is off the ground with ALL wheels, then block steering/leaning left/right/up/down.
+        if IsThisModelABoat(model) or IsThisModelAHeli(model) or IsThisModelAPlane(model) or not IsEntityInAir(veh) then
+            DisableControlAction(0, 59) -- leaning left/right
+            DisableControlAction(0, 60) -- leaning up/down
+        end
+    end
+end
```
Comment (BerkieBb) [inline client/main.lua]:
```suggestion
local function preventAirControl()
    if not cache.vehicle or IsEntityDead(cache.vehicle) then return end

    local model = GetEntityModel(cache.vehicle)
    -- If it's not a boat, plane or helicopter, and the vehicle is off the ground with ALL wheels, then block steering/leaning left/right/up/down.
    if IsThisModelABoat(model) or IsThisModelAHeli(model) or IsThisModelAPlane(model) or not IsEntityInAir(cache.vehicle) then return end

    DisableControlAction(0, 59, true) -- leaning left/right
    DisableControlAction(0, 60, true) -- leaning up/down
end
```
I'm adding the IsEntityDead as well, but I'm not sure if that works on vehicles

File: config.lua [hunk 6]
```diff
@@ -66,6 +66,7 @@ cfg = {
     limpModeMultiplier = 0.15,					-- The torque multiplier to use when vehicle is limping. Sane values are 0.05 to 0.25
 
     preventVehicleFlip = true,					-- If true, you can't turn over an upside down vehicle
+    preventAirControl  = true,					    -- If true, you can't AirControl vehicle
```
Comment (BerkieBb) [inline config.lua]:
```suggestion
    preventAirControl  = true,					-- If true, you can't AirControl vehicle
```

**Comment by BoltSpace** [comment]
@BerkieBb Made changes/suggestions do work. TY!

**Review by BerkieBb — APPROVED** [review]

**Review by KostaZx — CHANGES_REQUESTED** [review]
local veh = cache.vehicle

File: client/main.lua
Comment (KostaZx) [inline client/main.lua:1]: local veh = cache.vehicle

**Review by KostaZx — CHANGES_REQUESTED** [review]

File: client/main.lua [hunk 7]
```diff
@@ -471,6 +482,9 @@ if cfg.torqueMultiplierEnabled or cfg.preventVehicleFlip or cfg.limpMode then
             if cfg.preventVehicleFlip then
                 preventVehicleFlip()
             end
+            if cfg.preventAirControl then
+                preventAirControl()
+            end
         end
     end)
 end
```
Comment (solareon) [inline client/main.lua:490]: This is missing the check to put preventAirControl into starting the thread. Additionally if you could move this into a thread that only runs when in a vehicle that can improve performance.

File: client/main.lua
Comment (BerkieBb) [reply client/main.lua:1]: A little too late

File: client/main.lua (hunk 7 above)
Comment (BerkieBb) [reply client/main.lua:490]: This was supposed to be at line 473

File: client/main.lua
Comment (BoltSpace) [reply client/main.lua:1]: done

File: client/main.lua (hunk 7 above)
Comment (BoltSpace) [reply client/main.lua:490]: Added to 473

Comment (BoltSpace) [reply client/main.lua:490]: Replaced with solareon suggestions

**Comment by BoltSpace** [comment]
Added all changes as suggested.

**Review by solareon — APPROVED** [review]

**Review by KostaZx — APPROVED** [review]

**Pull request merged**